### ISSUE TITLE
CanOPEN import filter broken by known packets definitions

### ISSUE THREAD

**Issue by amund7** [amund7/CANBUS-Analyzer#10]
The latest log file from JWardell looks like this:

```
"CANopen Magic Ultimate 9.40 build 5290 Trace Recording, generated on 12/28/2018 1:11:04 AM"
"$FileVersion=5$"
"$TimestampMode=Absolute$"

"Message Number","Time (ms)","Time","Excel Time","Count","ID","Flags","Message Type","Node","Details","Process Data","Data (Hex)","Data (Text)","Data (Decimal)","Length","Raw Message"
"0","0.000","10:07:46:09.4848950'",43462.0382589288,"","0x2E1","","Default: PDO","","Default: TPDO 2 of Node 0x61 (97)","","03 DE 99 02 1B 1B E0 00 ",". Þ . . . . à . ","U:0 S:0","8","03 DE 99 02 1B 1B E0 00"
"1","0.640","10:07:46:09.4855350'",43462.0382589362,"","0x2D1","","Default: PDO","","Default: TPDO 2 of Node 0x51 (81)","","FF 00 ","ÿ . ","U:255 S:255","2","FF 00"
"2","1.749","10:07:46:09.4866440'",43462.038258949,"","0x201","","Default: PDO","","Default: RPDO 1 of Node 0x01 (1)","","E1 0A 20 C0 02 D8 D8 00 ","á .   À . Ø Ø . ","U:0 S:0","8","E1 0A 20 C0 02 D8 D8 00"
"3","3.626","10:07:46:09.4885210'",43462.0382589707,"","0x1F8","","Default: PDO","","Default: TPDO 1 of Node 0x78 (120)","","00 00 00 00 ",". . . . ","U:0 S:0","4","00 00 00 00"
"4","4.181","10:07:46:09.4890760'",43462.0382589771,"","0x379","","Default: PDO","","Default: RPDO 2 of Node 0x79 (121)","","C7 09 38 2F 28 3C 01 5E ","Ç . 8 / ( < . ^ ","U:0 S:0","8","C7 09 38 2F 28 3C 01 5E"
"5","4.394","10:07:46:09.4892890'",43462.0382589796,"","0x6C8","","","","","","01 00 00 00 00 00 00 00 ",". . . . . . . . ","U:0 S:0","8","01 00 00 00 00 00 00 00"
"6","5.162","10:07:46:09.4900570'",43462.0382589885,"","0x39A","","Default: PDO","","Default: TPDO 3 of Node 0x1A (26)","","94 87 57 A7 FF FE 7A 0F ",". . W § ÿ þ z . ","U:0 S:0","8","94 87 57 A7 FF FE 7A 0F"
"7","5.418","10:07:46:09.4903130'",43462.0382589915,"","0x129","","","","","","91 2C 20 20 00 20 FF 3F ",". ,     .   ÿ ? ","U:0 S:0","8","91 2C 20 20 00 20 FF 3F"
"8","5.589","10:07:46:09.4904840'",43462.0382589934,"","0x743","","Heartbeat/Node Guarding","0x43 (67)","Unknown","","","","","1","22"
"9","5.845","10:07:46:09.4907400'",43462.0382589964,"","0x393","","Default: PDO","","Default: TPDO 3 of Node 0x13 (19)","","09 80 63 00 00 00 00 00 ",". . c . . . . . ","U:0 S:0","8","09 80 63 00 00 00 00 00"
"10","8.917","10:07:46:09.4938120'",43462.038259032,"","0x132","","","","","Smooth Current:11.2Amps, Voltage132:384.63Volts, 132Current3:9.4Amps","3F 96 D8 26 C2 4D FF 0F ","? . Ø & Â M ÿ . ","U:0 S:0","8","3F 96 D8 26 C2 4D FF 0F"


```

and it breaks the reader at the last line, as a cell contains commas between quotes. In my code I just use Split(',') to separate the columns, not sure how to get around this

**Comment by Bokonon79** [comment]
I'll take a look at this, if for no other reason than to get my feet wet.

**Comment by scottstout** [comment]
If you want to take the easy route you can use a csv parsing library like this:

https://www.nuget.org/packages/CsvHelper/

Scott

Scott Stout

________________________________

**Comment by Bokonon79** [comment]
Resolved by PR #15.
